src/main/java/io/tiledb/java/api/Query.java

```diff
@@ -365,6 +427,93 @@ public synchronized Query setBuffer(String attr, NativeArray offsets, NativeArra
     return this;
   }
 
+  /**
+   * Sets a buffer for a variable-sized getAttribute.
+   *
+   * @param attr Attribute name
+   * @param offsets Offsets where a new element begins in the data buffer.
+   * @param buffer Buffer vector with elements of the attribute type.
+   * @param offsetElements The actual number of offset elements
+   * @param bufferElements The actual number of buffer elements
+   * @exception TileDBError A TileDB exception
+   */
+  public synchronized Query setBuffer(
+      String attr, NativeArray offsets, NativeArray buffer, int offsetElements, int bufferElements)
+      throws TileDBError {
+    Integer offsetSize = buffer.getSize();
+    Integer bufferSize = buffer.getSize();
+
+    if (offsetElements > offsetSize) {
+      throw new TileDBError(
+          "Number of offset elements requested exceeds the number of elements in allocated offset buffer: "
+              + offsetElements
+              + " > "
+              + offsetSize);
+    }
+    if (bufferElements > bufferSize) {
+      throw new TileDBError(
+          "Number of buffer elements requested exceeds the number of elements in allocated buffer"
+              + bufferElements
+              + " > "
+              + bufferSize);
+    }
+    if (attr.equals(tiledb.tiledb_coords())) {
+      throw new TileDBError("Cannot set coordinate buffer as variable sized.");
+    }
+
+    if (!offsets.getNativeType().equals(TILEDB_UINT64)) {
+      throw new TileDBError(
+          "Buffer offsets should be of getType TILEDB_UINT64 or Long. Found getType: "
+              + offsets.getNativeType());
+    }
+    if (offsetElements <= 0 || bufferElements <= 0) {
+      throw new TileDBError("Number of buffer elements must be >= 1");
+    }
+
+    // Type check the buffer native type matches the schema attribute type
+    try (ArraySchema schema = array.getSchema();
+        Attribute attribute = schema.getAttribute(attr)) {
+      Types.typeCheck(attribute.getType(), buffer.getNativeType());
+    }
+
+    uint64_tArray offsets_array = PointerUtils.uint64_tArrayFromVoid(offsets.toVoidPointer());
+    uint64_tArray offsets_array_size = new uint64_tArray(1);
+    uint64_tArray values_array_size = new uint64_tArray(1);
+
+    offsets_array_size.setitem(0, BigInteger.valueOf(offsets.getNBytes()));
+    values_array_size.setitem(0, BigInteger.valueOf(buffer.getNBytes()));
```

Comment:
You should set `bufferElements * buffer.getNativeTypeSize()` here so you only have to set it once instead of setting the size and updating it later. Same for offsets too.